subsys/testsuite/coverage/coverage.c

```diff
@@ -232,7 +232,34 @@ size_t gcov_to_gcda(uint8_t *buffer, struct gcov_info *info)
 	return buffer_write_position;
 }
 
-void dump_on_console_start(const char *filename)
+void gcov_reset_counts(struct gcov_info *info)
+{
+	struct gcov_fn_info *functions;
+	struct gcov_ctr_info *counters_per_func;
+	uint32_t iter_functions;
+	uint32_t iter_counts;
+	uint32_t iter_counter_values;
+
+	for (iter_functions = 0U;
```

Comment:
I think this traversal code is copy paste? Maybe extract it into it's own function?

Reply:
I am not sure it will make the code more readable. And I don't see any other place where it could be reused

Reply:
That would be kind of useful, even in terms of public API. The major benefit I think would be that external code could iterate over coverage data while keeping it in a single, private, location. This would enable coverage backends.

Reply:
There are 3 nested loops here and there are actions that are needed to be done on each level. Do you propose to define a traversal method that contains these 3 loops and accepts 3 callback functions? And each function will receive a context consisting of multiple variables?
I really don't see how to make it in a simple way that would improve things. Please suggest the exact api you are proposing

Reply:
That is kind of an interesting proposal. I hadn't thought about it with three callbacks, each one being applied to each level. 

What I was thinking is that mostly the "outer" two for loops are just finding out how to walk the counters. I was more thinking about a single function to operate on the specific counters themselves, which are really what the point of the code is in all of the places it is called. So the idea is, to provide something like a for each functionality for each counter.

Reply:
I think a seperate PR is acceptable or we could talk offline. A test would still be nice to see, but in the name of getting things moving I will accept as I don't see any obvious issues with the code.